lms/djangoapps/certificates/tests/test_queue.py

```diff
@@ -73,6 +78,55 @@ def test_no_create_action_in_queue_for_html_view_certs(self):
         self.assertEqual(certificate.status, CertificateStatuses.downloadable)
         self.assertIsNotNone(certificate.verify_uuid)
 
+    @ddt.data('honor', 'audit')
+    def test_add_cert_with_honor_certificates(self, mode):
+        """Test certificates generations for honor and audit modes."""
+        actual_body, certificate = self.assert_queue_response_with_different_modes(mode)
+        actual_body = json.loads(actual_body)
+        self.assertIn(
+            'certificate-template-{id.org}-{id.course}.pdf'.format(
+                id=self.course.id
+            ),
+            actual_body['template_pdf']
+        )
+        self.assertEqual(certificate.mode, mode)
+
+    @ddt.data('credit', 'verified')
+    def test_add_cert_with_verified_certificates(self, mode):
+        """Test if enrollment mode is verified or credit along with valid
+        software-secure verification than verified certificate should be generated.
+        """
+        actual_body, certificate = self.assert_queue_response_with_different_modes(mode)
+        actual_body = json.loads(actual_body)
+        self.assertIn(
+            'certificate-template-{id.org}-{id.course}-verified.pdf'.format(
+                id=self.course.id
+            ),
+            actual_body['template_pdf']
+        )
+        self.assertEqual(certificate.mode, "verified")
```

Comment:
@clintonb both test cases have different templates and certificate mode. In case of one helper method i have to add the if condition.